File: Samples/Android-Advanced/app/src/main/java/com/pushwoosh/sample/PushwooshSampleApp.java
package com.pushwoosh.sample;

import android.app.Application;
import android.util.Log;

import com.pushwoosh.Pushwoosh;
import com.pushwoosh.location.PushwooshLocation;
import com.pushwoosh.notification.LocalNotification;
import com.pushwoosh.tags.Tags;


public class PushwooshSampleApp extends Application {
	public static final String LTAG = "PushwooshSample";

	@Override
	public void onCreate() {
		super.onCreate();

		Pushwoosh.getInstance().registerForPushNotifications(result -> {
			if (result.isSuccess()) {
				Log.d(LTAG, "Successfully registered for push notifications with token: " + result.getData());
			} else {
				Log.d(LTAG, "Failed to register for push notifications:u " + result.getException().getMessage());
			}
		});

		PushwooshLocation.startLocationTracking();

		Pushwoosh.getInstance().sendTags(Tags.intTag("fav_number", 42));

		LocalNotification notification = new LocalNotification.Builder()
				.setMessage("Local notification")
				.setDelay(5)
				.build();

		Pushwoosh.getInstance().scheduleLocalNotification(notification);
	}
}

File: Samples/Android-Advanced-Kotlin/app/src/main/java/com/pushwoosh/sample/PushwooshSampleApp.kt
package com.pushwoosh.sample

import android.app.Application


import android.util.Log
import com.pushwoosh.Pushwoosh

import com.pushwoosh.location.PushwooshLocation
import com.pushwoosh.notification.LocalNotification
import com.pushwoosh.tags.Tags

class PushwooshSampleApp : Application() {
    companion object {
        const val LTAG = "PushwooshSample"
    }

    override fun onCreate() {
        super.onCreate()

        Pushwoosh.getInstance().registerForPushNotifications { result ->
            when(result.isSuccess) {
                true -> Log.d(LTAG, "Successfully registered for push notifications with token: " + result.data!!)
                false -> Log.d(LTAG, "Failed to register for push notifications:u " + result.exception!!.message)
            }
        }

        PushwooshLocation.startLocationTracking()

        Pushwoosh.getInstance().sendTags(Tags.intTag("fav_number", 42))

        val notification = LocalNotification.Builder()
                .setMessage("Local notification")
                .setDelay(5)
                .build()

        Pushwoosh.getInstance().scheduleLocalNotification(notification)
    }
}
